### PR TITLE
feat(nodestore): File-based nodestore backend

### PR DESCRIPTION
**Pull request by gggritso**
Output of a recent lightning talk. This is a `nodestore` backend that saves nodes as plaintext files in a folder. Very useful for debugging, testing edge cases, etc. I don't necessarily think we need to merge this, but it'd be a shame to never make the code available. I can either close this PR, or make whatever changes are necessary to merge this.

**To use:**

1. Set `SENTRY_NODESTORE = "sentry.nodestore.filesystem.FileSystemNodeStorage"`
2. Optionally set `SENTRY_NODESTORE_OPTIONS = {"path": "path-to-folder"}`
3. Run `sentry upgrade --with-nodestore`
4. Restart the Sentry app

### CODE REVIEW

File: src/sentry/nodestore/filesystem/backend.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+from sentry.nodestore.base import NodeStorage
+
+
+class FileSystemNodeStorage(NodeStorage):
+    """
+    A simple backend that saves each node as a file. Only appropriate for
+    debugging and development!
+    """
+
+    def __init__(self, path=None):
```
Comment (dashed) [inline src/sentry/nodestore/filesystem/backend.py:12]:
```suggestion
    def __init__(self, path=None):
        if not settings.DEBUG:
            raise ValueError("FileSystemNodeStorage should only be used in development!")
```

Comment (gggritso) [reply src/sentry/nodestore/filesystem/backend.py:12]: @dashed I like that, thanks!

**Review by wedamija — COMMENTED** [review]
Doesn't particularly bother me if we want to merge this, but does it have much benefit over the postgres backend?

File: src/sentry/nodestore/filesystem/backend.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
+import os
+
+from django.conf import settings
+
+from sentry.nodestore.base import NodeStorage
+
+
+class FileSystemNodeStorage(NodeStorage):
```
[the remark this answers is not in the document]
Comment (gggritso) [reply src/sentry/nodestore/filesystem/backend.py:10]: @dashed good call, thanks! Implemented a `cleanup` based on file creation time

**Comment by gggritso** [comment]
@wedamija I think it does! It's definitely a toy backend, but it's useful because unlike Postgres, it's plaintext. Postgres nodestore is gzipped and/or pickled, so it's really hard to manipulate. Plaintext is a lot easier to look at and edit (for testing edge cases, generating screenshots, etc.)

**Review by wedamija — APPROVED** [review]
> @wedamija I think it does! It's definitely a toy backend, but it's useful because unlike Postgres, it's plaintext. Postgres nodestore is gzipped and/or pickled, so it's really hard to manipulate. Plaintext is a lot easier to look at and edit (for testing edge cases, generating screenshots, etc.)

Totally forgot that we compress and pickle in the backend here. Seems good to me

**Review by dashed — APPROVED** [review]

**Pull request merged**